static/js/components/PersonalTab.js

```diff
@@ -78,6 +97,19 @@ export default class PersonalTab extends React.Component {
     );
   }
 
+  componentWillMount() {
+    const {
```

Comment:
Instead of resetting the `selectedProgram` state here, can you set it to null in `componentWillUnmount` to clean up once the component goes away? Then I think `getSelectedProgramId` will take care of populating the correct value